### PR TITLE
Fully specify behavior of unary operators

### PR DESCRIPTION
**Pull request by jonas-schievink**

### CODE REVIEW

File: src/expressions.rst [hunk 1]
```diff
@@ -1127,8 +1148,23 @@ The :t:`evaluation` of a :t:`negation expression` with a
 #. :dp:`fls_zdfgqky85r1f`
    The :t:`operand` is evaluated.
 
+#. :dp:`fls_CutpaCFCGHQs`
+   If the type of the :t:`operand` is an :t:`integer type`, the
+   :t:`negation expression` evaluates to the value of the operand, with its sign
```
Comment (kirtchev-adacore) [inline src/expressions.rst:1153]: the :t:`value` of the :t:`operand`

File: src/expressions.rst [hunk 2]
```diff
@@ -1127,8 +1148,23 @@ The :t:`evaluation` of a :t:`negation expression` with a
...
+   inverted.
+
+   #. :dp:`fls_MrvYkkEr4xKy`
+      If the result of the negation does not fit within the range of the
```
Comment (kirtchev-adacore) [inline src/expressions.rst:1157]: I would replace "the negation" with "the :t:`negation expression`".

File: src/expressions.rst [hunk 3]
```diff
@@ -1127,8 +1148,23 @@ The :t:`evaluation` of a :t:`negation expression` with a
...
+      :t:`operand` type, :t:`arithmetic overflow` occurs.
+
+#. :dp:`fls_B2eKGWaJhFKD`
+   If the type of the :t:`operand` is a :t:`floating-point type`, the
+   :t:`negation expression` evaluates to the value of the operand, with its sign
```
Comment (kirtchev-adacore) [inline src/expressions.rst:1162]: the :t:`value` of the :t:`operand`

**Review by kirtchev-adacore — CHANGES_REQUESTED** [review]

**Comment by kirtchev-adacore** [comment]
bors merge

**Comment by pietroalbini** [comment]
bors retry

**Pull request merged**